File: src/code-samples/deepagents/customization-interpreters.ts
// :snippet-start: customization-interpreters-js
import { createDeepAgent } from "deepagents";
import { createCodeInterpreterMiddleware } from "@langchain/quickjs";

const agent = createDeepAgent({
  model: "openai:gpt-5.4",
  middleware: [createCodeInterpreterMiddleware()],
});
// :snippet-end:

// :remove-start:
if (!agent) throw new Error("agent not created");
// :remove-end:

File: src/code-samples/deepagents/customization-interpreters.py
"""Customization: code interpreter middleware example."""

# :snippet-start: customization-interpreters-py
from deepagents import create_deep_agent
from langchain_quickjs import CodeInterpreterMiddleware

agent = create_deep_agent(
    model="anthropic:claude-sonnet-4-6",
    middleware=[CodeInterpreterMiddleware()],
)
# :snippet-end:

# :remove-start:
assert agent is not None
# :remove-end:
